sasmodels/models/core_shell_bicelle_elliptical.c

```diff
@@ -73,17 +51,17 @@ double
         const double cos_alpha = ( Gauss76Z[i] + 1.0 )/2.0;
         const double sin_alpha = sqrt(1.0 - cos_alpha*cos_alpha);
         double inner_sum=0;
-        double sinarg1 = qq*halfheight*cos_alpha;
-        double sinarg2 = qq*(halfheight+facthick)*cos_alpha;
+        double sinarg1 = q*halfheight*cos_alpha;
+        double sinarg2 = q*(halfheight+thick_face)*cos_alpha;
         si1 = sas_sinx_x(sinarg1);
         si2 = sas_sinx_x(sinarg2);
         for(int j=0;j<76;j++) {
             //76 gauss points for the inner integral (WAS 20 points,so this may make unecessarily slow, but playing safe)
             //const double beta = ( Gauss76Z[j]*(vbj-vaj) + vaj + vbj )/2.0;
             const double beta = ( Gauss76Z[j] +1.0)*M_PI_2;
             const double rr = sqrt(rA - rB*cos(beta));
-            double besarg1 = qq*rr*sin_alpha;
-            double besarg2 = qq*(rr+radthick)*sin_alpha;
+            double besarg1 = q*rr*sin_alpha;
+            double besarg2 = q*(rr+thick_rim)*sin_alpha;
```

Comment:
I think the thick_rim part is very likely OK. Though the core is elliptical, the rim is constant thickness, thus if we make the whole bicelle the same sld, the comparison with elliptical_cylinder or core_shell_bicelle has to be done with a different axial ratio to compensate. 
Need to check whether the 2d version that you sorted for me does actually do this.

If the cos(beta) bit is wrong then it is also wrong in elliptical cylinder.

Have renamed rA to r2A to remind us that it is a length^2